astropy/io/misc/hdf5.py

```diff
@@ -213,6 +213,26 @@ def _encode_mixins(tbl):
 
     return encode_tbl
 
+def check_column_encoding(table):
+    """
+    Table with native py3 strings can't be stored in a hdf5 format so if
+    table has py3 strings then a copy of table is made and stored in temp.
+    the copy stored in temp has columns in bytes unicode.
+    """
+    from astropy.table import Table
+    col = []
+    temp = Table()
+    for column in table.colnames:
+        if hasattr((table.columns[column]), 'dtype'):
+            if str(table.columns[column].dtype.kind) == 'U':
+                col.append(column)
```

Comment:
This loop could probably be simplified :
```
columns = [name for name, col in table.columns.items() if col.dtype.kind == 'U']
```

Reply:
okay I will make the changes.